### PR TITLE
Admin bro

### PR DESCRIPTION
**Pull request by zatkiller**
Current AdminBro integration

Currently we need to use their buildAuthenticatedRouter to use their built in log in page for the admin side. However the authentication process for this login page can still be implemented by us.

Need a auth to differentiate from admin and normal users

AdminBro does not play well with `gts`'s `tsconfig.json`. Copied `gts` `tsconfig.json` into our own tsconfig and set `noImplicitAny` to false for typescript to compile. Removed `strict: true` along the way.

### CODE REVIEW

File: src/middleware/adminAuth.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const ADMIN = {
+  email: 'test@example.com',
+  password: 'password',
+};
+
+const adminAuth = (email, password) => {
```
Comment (wenhaogoh) [inline src/middleware/adminAuth.ts]: Just a minor issue, guard clauses should catch errors and the final return statement should be the success scenario: https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent

File: src/middleware/adminAuth.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  if (ADMIN.password === password && ADMIN.email === email) {
```
Comment (wenhaogoh) [inline src/middleware/adminAuth.ts]: Same here.

**Review by wenhaogoh — APPROVED** [review]
Just some nits to pick. Otherwise, LGTM!

File: .adminbro/.entry.js [hunk 3]
```diff
@@ -0,0 +1 @@
+AdminBro.UserComponents = {}
```
Comment (ivantjh) [inline .adminbro/.entry.js]: I think this is auto-generated, help put the whole folder into `.gitignore`

**Review by ivantjh — CHANGES_REQUESTED** [review]

File: src/middleware/adminAuth.ts (hunk 1 above)
Comment (zatkiller) [reply src/middleware/adminAuth.ts]: Fixed. Will merge if this is correct way to use guard clauses

**Review by wenhaogoh — APPROVED** [review]
LGTM

**Pull request merged**